cmake/kconfig.cmake

```diff
@@ -63,6 +63,20 @@ set(ENV{ARCH_DIR}   ${ARCH_DIR})
 set(ENV{TOOLCHAIN_KCONFIG_DIR} "${TOOLCHAIN_KCONFIG_DIR}")
 set(ENV{EDT_PICKLE} ${EDT_PICKLE})
 
+# Export each `ZEPHYR_<module>_MODULE_DIR` to Kconfig.
+# This allows Kconfig files to refer relative from a modules root as:
+# source "$(ZEPHYR_FOO_MODULE_DIR)/Kconfig"
+foreach(module_name ${ZEPHYR_MODULE_NAMES})
+  string(TOUPPER ${module_name} MODULE_NAME_UPPER)
+  list(APPEND
+       ZEPHYR_KCONFIG_MODULES_DIR
+       "ZEPHYR_${MODULE_NAME_UPPER}_MODULE_DIR=${ZEPHYR_${MODULE_NAME_UPPER}_MODULE_DIR}"
+  )
+  set(ENV{ZEPHYR_${MODULE_NAME_UPPER}_MODULE_DIR}
+      ${ZEPHYR_${MODULE_NAME_UPPER}_MODULE_DIR}
+  )
```

Comment:
Why is this necessary to set in the main build environment? I would guess adding it to the environments of each kconfig_target would be enough. We have too many environment variables already IMO.

Reply:
(This one is not a blocker, just asking if we can avoid adding extra environment variables to the cmake process.)

Reply:
I completely agree, however this commit follows the existing lines here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/kconfig.cmake#L51-L64
and is currently needed because an `execute_process()` here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/kconfig.cmake#L213-L216

Cleaning up this is outside scope of this PR, but con follow in a dedicated PR covering all above mentioned lines.

Reply:
> and is currently needed because an `execute_process()` here:
> https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/kconfig.cmake#L213-L216

Got it. Thanks for the reference.

Reply:
included a cleanup commit.